File: public/examples/php/advanced/01_functions.php
<?php
// PHP Advanced: Functions

function greet($name, $greeting = "Hello")
{
    return "$greeting, $name!";
}

echo greet("Bob") . "\n";

// Typed parameters (PHP 7+)
function addNumbers(int $a, int $b): int
{
    return $a + $b;
}

echo "Sum: " . addNumbers(5, 10) . "\n";

// Anonymous function
$square = function ($n) {
    return $n * $n;
};
echo "Square of 4: " . $square(4) . "\n";
?>

File: public/examples/python/advanced/01_functions.py
# Python Advanced: Functions

# Standard function
def greet(name, greeting="Hello"):
    return f"{greeting}, {name}!"

print(greet("Bob"))
print(greet("Alice", greeting="Hi"))

# Args and Kwargs
def sum_all(*args, **kwargs):
    s = sum(args)
    label = kwargs.get("label", "Sum")
    return f"{label}: {s}"

print(sum_all(1, 2, 3, 4, label="Total"))

# Lambda function
square = lambda x: x * x
print("Square of 5:", square(5))
